### PR TITLE
Remove `WorldLocationType` model

### PR DESCRIPTION
**Pull request by brucebolt**
We are currently using the `world_location_type_id`  column to act as a key for the entries in the WorldLocationType model, which only contains 2 entries and behaves exactly like an enum.

Therefore we can actually use an enum here, which allows us to reduce a lot of code complexity.

The key numbering is maintained in line with the existing WorldLocationType entries.

This also automatically gives us some new helper methods on the WorldLocation model: `world_location_types`, `world_location?` and `international_delegation?`.

[Trello card](https://trello.com/c/32SGcOpZ)

⚠️ This repo is Continuously Deployed: make sure you [follow the guidance](https://docs.publishing.service.gov.uk/manual/development-pipeline.html#merge-your-own-pull-request) ⚠️

### CODE REVIEW

File: app/models/world_location.rb [hunk 1]
```diff
@@ -104,7 +104,7 @@ def to_s
   end
 
   def self.all_by_type
-    ordered_by_name.group_by(&:world_location_type)
+    ordered_by_name.in_order_of(:world_location_type, %w[world_location international_delegation]).group_by(&:world_location_type)
```
Comment (BeckaL) [inline app/models/world_location.rb:107]: Question out of interest more than anything - is it possible to re-use the enum here to define the ordering?

File: app/models/world_location_type.rb [hunk 2]
```diff
@@ -1,24 +0,0 @@
-class WorldLocationType
```
Comment (BeckaL) [inline app/models/world_location_type.rb:1]: 👍 there was so much unnecessary stuff in here! Really nice to get rid of this stuff

File: app/models/world_location.rb (hunk 1 above)
Comment (brucebolt) [reply app/models/world_location.rb:107]: I did wonder if we could do that, but decided against it as any future additions to the enum may not necessarily appear in the order we want them to be listed on the website.

File: app/models/world_location_type.rb [hunk 3]
```diff
@@ -1,24 +0,0 @@
...
-  include ActiveRecordLikeInterface
-
-  attr_accessor :id, :name, :sort_order, :key
-
-  def slug
-    name.downcase.gsub(/[^a-z]+/, "-")
```
Comment (BeckaL) [inline app/models/world_location_type.rb:7]: Leaving half a thought here before I have a bit more time to delve into this - was this slug used anywhere except in the find_by_slug below (and was that used anywhere)? Just want to make sure that this isn't somehow used to generate the slug of the overall document (I think not but haven't fully looked into this yet)

Comment (brucebolt) [reply app/models/world_location_type.rb:7]: There's a `slug` database field in the `world_locations` table, so Rails automatically generates a `slug` method for these objects.

**Review by BeckaL — APPROVED** [review]
All looks sensible to me, and a super nice simplification of the code. ⭐ 

Originally I was thinking we'd replace the numeric ids in the database with literally the text "world_location" or "international_delegation" but this is a way nicer / less sketchy solution. 

Hopefully it'll be a helpful thing to have in place when we do further refactoring on world locations / international delegations, but even if we don't get to that this is a big improvement 👍

**Pull request merged**